src/status_im/contexts/communities/actions/addresses_for_permissions/view.cljs

```diff
@@ -1,55 +1,76 @@
             [status-im.contexts.communities.actions.addresses-for-permissions.style :as style]
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
 (defn- account-item
-  [item]
-  [rn/view
-   {:style style/account-item-container}
-   [quo/account-avatar (assoc item :size 32)]
-   [rn/view
-    [quo/text
-     {:size   :paragraph-1
-      :weight :semi-bold}
-     (:name item)]
-    [quo/address-text
-     {:address (:address item)
-      :format  :short}]]])
+  [item _ _ selected-addresses]
+  [quo/account-permissions
+   {:account         {:name                (:name item)
+                      :address             (:address item)
+                      :emoji               (:emoji item)
+                      :customization-color (:customization-color item)}
+    :token-details   []
+    :checked?        (contains? @selected-addresses (:address item))
+    :on-change       (fn [checked?]
+                       (if checked?
+                         (swap! selected-addresses conj (:address item))
+                         (swap! selected-addresses disj (:address item))))
+    :container-style {:margin-bottom 8}}])
 
 (defn view
   []
   (let [{id :community-id}          (rf/sub [:get-screen-params])
         {:keys [name color images]} (rf/sub [:communities/community id])
-        accounts                    (rf/sub [:wallet/accounts-with-customization-color])]
-    [rn/safe-area-view {:style style/container}
+        accounts                    (rf/sub [:wallet/accounts-with-customization-color])
+        selected-addresses          (reagent/atom (rf/sub [:communities/addresses-for-permissions]))]
+    (fn []
+      [rn/safe-area-view {:style style/container}
+       [quo/drawer-top
+        {:type                :context-tag
+         :title               (i18n/label :t/addresses-for-permissions)
+         :community-name      name
+         :button-icon         :i/info
+         :on-button-press     not-implemented/alert
+         :community-logo      (get-in images [:thumbnail :uri])
+         :customization-color color}]
 
-     [quo/drawer-top
-      {:type                :context-tag
-       :title               (i18n/label :t/addresses-for-permissions)
-       :community-name      name
-       :button-icon         :i/info
-       :on-button-press     not-implemented/alert
-       :community-logo      (get-in images [:thumbnail :uri])
-       :customization-color color}]
+       [rn/flat-list
+        {:render-fn               account-item
+         :render-data             selected-addresses
+         :content-container-style {:padding 20}
+         :key-fn                  :address
+         :data                    accounts}]
 
-     [rn/flat-list
-      {:render-fn               account-item
-       :content-container-style {:padding 20}
-       :key-fn                  :address
-       :data                    accounts}]
+       (when (empty? @selected-addresses)
+         [rn/view
```

Comment:
Please ignore the inaccurate colors, this is a temporary implementation. I will replace this with drawer/bottom-actions component after updating that component.

https://github.com/status-im/status-mobile/issues/15006